src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -964,28 +965,31 @@ public void CanBindSingleElementToCollection()
             Assert.Equal(1, nestedAsArray.Length);
         }
 
-        [Fact]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public void CannotBindSingleElementToCollectionWhenSwitchSet()
         {
-            AppContext.SetSwitch("Microsoft.Extensions.Configuration.BindSingleElementsToArray", false);
-
-            var dic = new Dictionary<string, string>
+            RemoteExecutor.Invoke(() =>
             {
-                {"MyString", "hello world"},
-                {"Nested:Integer", "11"},
-            };
+                AppContext.SetSwitch("Microsoft.Extensions.Configuration.BindSingleElementsToArray", false);
 
-            var configurationBuilder = new ConfigurationBuilder();
-            configurationBuilder.AddInMemoryCollection(dic);
-            IConfiguration config = configurationBuilder.Build();
+                var dic = new Dictionary<string, string>
+                {
+                    {"MyString", "hello world"},
+                    {"Nested:Integer", "11"},
+                };
 
-            var stringArr = config.GetSection("MyString").Get<string[]>();
-            Assert.Null(stringArr);
+                var configurationBuilder = new ConfigurationBuilder();
+                configurationBuilder.AddInMemoryCollection(dic);
+                IConfiguration config = configurationBuilder.Build();
 
-            var stringAsStr = config.GetSection("MyString").Get<string>();
-            Assert.Equal("hello world", stringAsStr);
+                var stringArr = config.GetSection("MyString").Get<string[]>();
+                Assert.Null(stringArr);
+
+                var stringAsStr = config.GetSection("MyString").Get<string>();
+                Assert.Equal("hello world", stringAsStr);
 
-            AppContext.SetSwitch("Microsoft.Extensions.Configuration.BindSingleElementsToArray", true);
+                AppContext.SetSwitch("Microsoft.Extensions.Configuration.BindSingleElementsToArray", true);
```

Comment:
Don't need to set it to `true` at the end, since the process will exit after this.